pkg/demoinfocs/common/common.go

```diff
@@ -65,11 +65,12 @@ func (h *DemoHeader) FrameTime() time.Duration {
 // GrenadeProjectile is a grenade thrown intentionally by a player. It is used to track grenade projectile
 // positions between the time at which they are thrown and until they detonate.
 type GrenadeProjectile struct {
-	Entity         st.Entity
-	WeaponInstance *Equipment
-	Thrower        *Player     // Always seems to be the same as Owner, even if the grenade was picked up
-	Owner          *Player     // Always seems to be the same as Thrower, even if the grenade was picked up
-	Trajectory     []r3.Vector // List of all known locations of the grenade up to the current point
+	Entity           st.Entity
+	WeaponInstance   *Equipment
+	Thrower          *Player       // Always seems to be the same as Owner, even if the grenade was picked up
+	Owner            *Player       // Always seems to be the same as Thrower, even if the grenade was picked up
+	Trajectory       []r3.Vector   // List of all known locations of the grenade up to the current point
+	TrajectoryTiming []PointInTime // List of points in time for known locations of the grenade up to the current point
```

Comment:
Maybe we should deprecate `Trajectory` and add a new field `Trajectory2` of type `TrajectoryEntry` and then we can add `r3.Vector` and Frame + Time to that struct?

Reply:
Yeah sure, that could work as well!

Reply:
Something like this?

Reply:
looks great!